carmajava/signal_plugin/src/main/java/gov/dot/fhwa/saxton/carma/signal_plugin/TrafficSignalPlugin.java

```diff
@@ -390,15 +411,103 @@ static protected SpatMessage convertSpatMessage(IntersectionData data) {
         return cnvSpat;
     }
 
-    /**
-     * Helper function to trigger a new plan from arbitrator to accommodate for Glidepath planning
-     * 
-     * @param availability The availability to set when calling this replan
-     */
-    private void triggerNewPlan(boolean availability) {
-        log.info("Requesting replan with availability: " + availability);
-        setAvailability(availability);
-        pluginServiceLocator.getArbitratorService().requestNewPlan();
+    private void setPlan(List<Node> plan, double startingDowntrack) {
+        synchronized (currentPlan) {
+            currentPlan.set(plan);
+            planStartingDowntrack.set(startingDowntrack);
+        }
+    }
+
+    @Override
+    public void triggerNewPlan(final boolean availability) {
+        log.info("Trying to request replan with availability: " + availability);
+        if (replanning.compareAndSet(false, true)) {
+
+            // Just return if the availability is false
+            if (!availability) {
+                setAvailability(availability);
+                pluginServiceLocator.getArbitratorService().requestNewPlan();
+                replanning.set(false);
+                return;
+            }
+
+            // Generate plan in new thread
+            Thread planningThread = new Thread(new Runnable() {
+
+                @Override
+                public void run() {
+                    log.info("Attempting Ead Plan");
+                    // Planning
+                    double currentDowntrack = pluginServiceLocator.getRouteService().getCurrentDowntrackDistance();
+                    double speedLimit = pluginServiceLocator.getRouteService().getSpeedLimitAtLocation(currentDowntrack).getLimit();
+                    DataElementHolder state = null;
+                    List<Node> eadResult = null;
+                    
+                    for (int i = 0; i < 3; i++) {
```

Comment:
This is probably worth adding a config value for, or at least refactoring into a constant `MAX_PLAN_RETRIES` variable.